pandas/tests/groupby/test_groupby.py

```diff
@@ -3189,6 +3189,15 @@ def test_depr_get_group_len_1_list_likes(test_series, kwarg, value, name, warn):
     tm.assert_equal(result, expected)
 
 
+def test_groupby_ngroup_with_nan():
+    # GH#50100
+    df = DataFrame({"a": Categorical([np.nan]), "b": [1]})
+    g = df.groupby(["a", "b"], dropna=False, observed=False).ngroup()
+    result = g.iloc[0]
+    expected = 0
+    tm.assert_equal(result, expected)
```

Comment:
```suggestion
    assert result == expected
```

Reply:
Can we go the opposite way here: test the full result of ngroup.

```
result = df.groupby(["a", "b"], dropna=False, observed=False).ngroup()
expected = ...
tm.assert_series_equal(result, expected)
```

Reply:
Agreed, Could you convert the test to this @josemayer

Reply:
Sure!